tests/app/questionnaire/routing/test_operators.py

```diff
@@ -314,6 +315,64 @@ def test_operation_date(date_string: str, offset):
     assert operator.evaluate(operands) == expected_result
 
 
+@pytest.mark.parametrize(
+    "offset, expected_result",
+    [
+        # Last Thursday (Yesterday from reference date)
+        ({"day_of_week": "THURSDAY"}, datetime(year=2020, month=12, day=31)),
+        # Last Saturday
+        ({"day_of_week": "SATURDAY"}, datetime(year=2020, month=12, day=26)),
+        # Last Week Thursday
+        (
+            {"day_of_week": "THURSDAY", "days": -7},
+            datetime(year=2020, month=12, day=24),
+        ),
+        # Last Week Saturday (Same as Last Saturday since reference date is a Friday)
+        (
+            {"day_of_week": "SATURDAY", "days": -7},
+            datetime(year=2020, month=12, day=26),
+        ),
+        # Next Thursday / Next Week Thursday
+        ({"day_of_week": "THURSDAY", "days": 7}, datetime(year=2021, month=1, day=7)),
+        # Next Saturday (Tomorrow from reference date)
+        ({"day_of_week": "SATURDAY", "days": 7}, datetime(year=2021, month=1, day=2)),
+        # Next Week Saturday
+        ({"day_of_week": "SATURDAY", "days": 14}, datetime(year=2021, month=1, day=9)),
+        # 1 full week Thursday
+        ({"day_of_week": "THURSDAY", "days": 14}, datetime(year=2021, month=1, day=14)),
```

Comment:
Do we need all these combination? I find this a bit strange "Next Thursday / Next Week Thursday" perhaps just one, because reference date is Friday that makes it New week Thursday I guess

Reply:
`Next Thursday / Next Week Thursday` added both to make it clearer, they are not different for this test case where we are on a Friday.